apis/apiextensions/fn/proto/v1alpha1/run_function.proto

```diff
@@ -39,22 +39,11 @@ enum ImagePullPolicy {
   IMAGE_PULL_POLICY_NEVER = 3;
 }
 
-// ImagePullAuth configures authentication to a remote OCI registry.
-// It corresponds to go-containerregistry's AuthConfig type.
-// https://pkg.go.dev/github.com/google/go-containerregistry@v0.11.0/pkg/authn#AuthConfig
-message ImagePullAuth {
-  string username = 1;
-  string password = 2;
-  string auth = 3;
-  string identity_token = 4;
-  string registry_token = 5;
-}
-
 // ImagePullConfig configures how a Composition Function container should be
 // pulled from a remote OCI registry.
 message ImagePullConfig {
   ImagePullPolicy pull_policy = 1;
-  ImagePullAuth auth = 2;
+  repeated string pull_secrets = 2;
 }
```

Comment:
I don't think we can make this change. We need the Crossplane core pod, not `xfn` to resolve pull secrets into image pull auth credentials. This is because we expect other Composition Function runners will exist, and those runners won't always be running in the same Kubernetes cluster as the Crossplane pod (they might not be running in Kubernetes at all).